include/maths/common/CSolvers.h

```diff
@@ -701,7 +714,7 @@ class MATHS_COMMON_EXPORT CSolvers {
                 return true;
             }
 
-            if (fa * fc > 0.0) {
+            if (std::signbit(fa) == std::signbit(fc)) {
```

Comment:
As above, this is slightly different to before if one or both variables is zero.

Reply:
As above neither variable can be 0 here.